ui/lib/core/addon/components/certificate-card.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+
+/**
+ * @module CertificateCard
+ * The CertificateCard component receives data and optionally receives a boolean declaring if that data is meant to be in PEM
+ * Format. It renders using the <HDS::Card::Container>. To the left there is a certificate icon. In the center there is a label
+ * which says which format (PEM or DER) the data is in. Below the label is the truncated data. To the right there is a copy
+ * button to copy the data.
+ *
+ * @example
+ * ```js
+ *  <CertificateCard @data={{value}} @isPem={{true}} />
+ * ```
+ * @param {string} data - the data to be displayed in the component (usually in PEM or DER format)
+ * @param {boolean} isPem - optional argument for if the data is required to be in PEM format (and should thus have the PEM Format label)
+ */
+
+export default class CertificateCardComponent extends Component {
+  // Returns the format the data is in: PEM, DER, or no format if no data is provided
+  get format() {
+    if (!this.args.data) return '';
+
+    let value;
+    if (typeof this.args.data === 'object') {
+      value = this.args.data[0];
+    } else {
+      value = this.args.data;
+    }
+
+    if (value.substring(0, 11) === '-----BEGIN ' || this.args.isPem === true) {
+      return 'PEM Format';
+    }
+    return 'DER Format';
+  }
+}
```

Comment:
Is there knowledge of what the format should be based on the attribute that needs to use this component or does it always have to be parsed from the certificate data? If it's known then I would suggest making `format` a component arg and default it to `PEM Format` and then you could get rid of the `isPem` arg.

```js
get format() {
  return this.args.format || 'PEM Format'; // or whatever the sane default might be
}
```

Reply:
I agree that that would be a much more intuitive and cleaner way to get the format! Unfortunately, if I'm not mistaken there isn't knowledge of what format the data is in when it is passed to `<CertificateCard>`. In most cases the user is choosing what format they want in a form field, but the resulting certificate isn't always the format the user selected due to a backend discrepancy ([more in this Slack thread](https://hashicorp.slack.com/archives/C0386B7KPHR/p1689718377604339)). Therefore I think the only way to get the format is parsing the data, because even if we passed in the user-chosen format to `<CertificateCard>`, it might not be the actual format the data get rendered in on the backend